deployment/cdk/opensearch-service-migration/lib/network-stack.ts

```diff
@@ -229,8 +230,11 @@ export class NetworkStack extends Stack {
         }
 
         // Create Source SSM Parameter
-        if (props.sourceClusterEndpoint) {
-            createMigrationStringParameter(this, props.sourceClusterEndpoint, {
+        if (props.sourceClusterEndpoint || props.sourceClusterDisabled) {
+            // The migration console stack depends on this parameter, even if the source cluster is disabled, so
```

Comment:
Why does the Migration Console stack depend on this, aren't we already controlling getting the SSM parameter with the sourceClusterDisabled paramter we are passing to the Migration Console stack?

Reply:
Apart from this though it looks like the RFS and FetchMigration stacks use the SOURCE_CLUSTER_ENDPOINT SSM parameter. RFS looks like it actually doesn't use it and can be removed. And we should probably just include Fetch in the conditional to throw an error if sourceClusterDisabled is provided.

Reply:
> Why does the Migration Console stack depend on this, aren't we already controlling getting the SSM parameter with the sourceClusterDisabled paramter we are passing to the Migration Console stack?


Yes, butttt CDK is only creating cloudformation and the cloudformation template includes it in the "parameter" section regardless of any conditionality around whether it's used. So as far as I can tell, we need to create this parameter if there are any stacks that _might_ depend on it, even if they don't actually.

Reply:
^ update, I was fortunately wrong about this!